src/expr.rs

```diff
@@ -245,8 +245,8 @@ fn format_expr(expr: &ast::Expr,
         ast::ExprKind::InlineAsm(..) => {
             wrap_str(context.snippet(expr.span), context.config.max_width, shape)
         }
-        // FIXME(#1537)
-        ast::ExprKind::Catch(..) => unimplemented!(),
+        // 6 = `catch `
+        ast::ExprKind::Catch(ref block) => block.rewrite(&context, try_opt!(shape.sub_width(6))),
```

Comment:
Yeah, I think `do catch` should follow `unsafe`. I don't think we should ever line-break between `do` and `catch`